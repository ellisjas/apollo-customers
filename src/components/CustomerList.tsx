import React, { useState, useEffect } from 'react'
import { ZellerCustomers } from '../API'
import styled from 'styled-components'
import Header from './Header'

const Container = styled.div`
  border-top: 1px solid lightgrey;
  border-bottom: 1px solid lightgrey;
  padding: 10px 0;
  margin: 30px 0;
`

interface CustomerListProps {
  customers: ZellerCustomers[]
  role: string
}

const CustomerList: React.FC<CustomerListProps> = ({ customers }) => {
  return (
    <>
      {customers.map((customer: ZellerCustomers) => {
        return (
          <div key={customer.id}>
            <h3>{customer.name}</h3>
            <p>{customer.role}</p>
          </div>
        )
      })}
    </>
  )
}

export default CustomerList
