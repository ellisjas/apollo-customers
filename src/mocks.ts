import { ZellerCustomers } from "./API";

export const customer: ZellerCustomers = {
  id: '3457356745',
  email: 'jasmine@example.com',
  name: 'Jasmine',
  role: 'ADMIN',
  __typename: 'ZellerCustomers',
}

export const customerList: ZellerCustomers[] = [
  customer,
  {
    id: '4563456',
    email: 'mark@example.com',
    name: 'Mark',
    role: 'MANAGER',
    __typename: 'ZellerCustomers',
  },
  {
    id: '34574567',
    email: 'tim@example.com',
    name: 'Tim',
    role: 'MANAGER',
    __typename: 'ZellerCustomers',
  },
  {
    id: '467456745',
    email: 'edie@example.com',
    name: 'Edie',
    role: 'MANAGER',
    __typename: 'ZellerCustomers',
  },
  {
    id: '23542345',
    email: 'julian@example.com',
    name: 'Julian',
    role: 'ADMIN',
    __typename: 'ZellerCustomers',
  }
]