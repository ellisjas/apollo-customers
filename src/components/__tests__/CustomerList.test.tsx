import React from 'react'
import { render } from '@testing-library/react'
import CustomerList from '../CustomerList'
import { customerList } from '../../mocks'

describe('components: <CustomerList />', () => {
  it('renders', () => {
    const { container } = render(<CustomerList customers={customerList} />)

    expect(container).toMatchSnapshot()
  })
})
