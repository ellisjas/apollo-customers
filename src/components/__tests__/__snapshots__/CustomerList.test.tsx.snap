// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components: <CustomerList /> renders 1`] = `
<div>
  <div
    class="sc-hLBbgP gXRIlG"
  >
    <div
      class="sc-eDvSVe cslRUZ"
    >
      J
    </div>
    <div
      class="sc-jSUZER kalImh"
    >
      <h3
        class="sc-gswNZR dejOgB"
      >
        Jasmine
      </h3>
      <p
        class="sc-dkrFOg jINReh"
      >
        Admin
      </p>
    </div>
  </div>
  <div
    class="sc-hLBbgP gXRIlG"
  >
    <div
      class="sc-eDvSVe cslRUZ"
    >
      M
    </div>
    <div
      class="sc-jSUZER kalImh"
    >
      <h3
        class="sc-gswNZR dejOgB"
      >
        Mark
      </h3>
      <p
        class="sc-dkrFOg jINReh"
      >
        Manager
      </p>
    </div>
  </div>
  <div
    class="sc-hLBbgP gXRIlG"
  >
    <div
      class="sc-eDvSVe cslRUZ"
    >
      T
    </div>
    <div
      class="sc-jSUZER kalImh"
    >
      <h3
        class="sc-gswNZR dejOgB"
      >
        Tim
      </h3>
      <p
        class="sc-dkrFOg jINReh"
      >
        Manager
      </p>
    </div>
  </div>
  <div
    class="sc-hLBbgP gXRIlG"
  >
    <div
      class="sc-eDvSVe cslRUZ"
    >
      E
    </div>
    <div
      class="sc-jSUZER kalImh"
    >
      <h3
        class="sc-gswNZR dejOgB"
      >
        Edie
      </h3>
      <p
        class="sc-dkrFOg jINReh"
      >
        Manager
      </p>
    </div>
  </div>
  <div
    class="sc-hLBbgP gXRIlG"
  >
    <div
      class="sc-eDvSVe cslRUZ"
    >
      J
    </div>
    <div
      class="sc-jSUZER kalImh"
    >
      <h3
        class="sc-gswNZR dejOgB"
      >
        Julian
      </h3>
      <p
        class="sc-dkrFOg jINReh"
      >
        Admin
      </p>
    </div>
  </div>
</div>
`;
